shaders/IronChunk.gdshader

```diff
@@ -1,30 +1,42 @@
 shader_type spatial;
 render_mode depth_prepass_alpha;
 
-uniform sampler2D albedoTexture : source_color;
+uniform sampler2D fullTexture : source_color;
+uniform sampler2D depletedTexture : source_color;
 uniform sampler2D normalTexture;
 
+// Increase to fade the iron out of the rock
+uniform float depletion : hint_range(0, 1);
+
 uniform sampler2D dissolveTexture : source_color;
 uniform float dissolveValue : hint_range(0, 1);
-
 uniform float outlineWidth;
 uniform vec4 growColor : source_color;
 
 void fragment() {
-    vec4 mainTex = texture(albedoTexture, UV);
+    vec4 fullTex = texture(fullTexture, UV);
+    vec4 depletedTex = texture(depletedTexture, UV);
+    vec3 mixcolor = (fullTex.rgb * (1.f - depletion)) + (depletedTex.rgb * depletion);
+    
+    //float advance = clamp(depletion + sin(TIME / 1.f), 0, 1);
+    
+    //vec3 mixcolor = (fullTex.rgb * (1.f - advance)) + (depletedTex.rgb * advance);
+    
+    float metal = clamp(mixcolor.r - mixcolor.b, 0, 1);
+    
     vec4 normalMap = texture(normalTexture, UV);
+    
+    // Handle disolve animation
     vec4 dissolveTex = texture(dissolveTexture, UV);
-
-    float cutoff = dot(dissolveTex.rgb, vec3(0.382, 0.382, 0.382)) -
-        float(-0.487 + clamp(dissolveValue, 0, 1));
-
+    float cutoff = dot(dissolveTex.rgb, vec3(0.29, 0.29, 0.29)) -
+        float(-0.55 + clamp(dissolveValue, 0, 1));
```

Comment:
I'd appreciate keeping the blank lines around this complex code line to keep readability better.